From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MartijnMuijsers <martijnmuijsers@live.nl>
Date: Wed, 30 Nov 2022 13:00:40 +0100
Subject: [PATCH] 5 second TPS average

License: MIT (https://opensource.org/licenses/MIT)

This patch is based on the following patch:
"Add 5 second tps average in /tps"
By: William Blake Galbreath <blake.galbreath@gmail.com>
As part of: Purpur (https://github.com/PurpurMC/Purpur)
Licensed under: MIT (https://opensource.org/licenses/MIT)

diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index f502bf2d511df86bc3c0b6a28682484e07a14120..ed787f736e36c08e10dfdca204011381123f1949 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -1751,6 +1751,15 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
 
     // Gale start - specific interval TPS API
 
+    // Gale start - Purpur - 5 second TPS average
+    /**
+     * Gets the average server TPS over the last 5 seconds
+     *
+     * @return the average server TPS over the last 5 seconds
+     */
+    double get5SecondTPSAverage();
+    // Gale end - Purpur - 5 second TPS average
+
     /**
      * Gets the average server TPS over the last 1 minute
      *
